### PR TITLE
KNDP-36: stack operator

### PR DESCRIPTION
**Pull request by Timex1854**

### CODE REVIEW

File: packages/operator/main.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+				w.WriteHeader(http.StatusOK)
+				return
+			}
+
+			next.ServeHTTP(w, r)
+		})
+	}
+
+	go func() {
+		http.HandleFunc("/api/objects", func(w http.ResponseWriter, r *http.Request) {
```
Comment (web7evgheni) [inline packages/operator/main.go:123]: @Timex1854 Why not "/api/stacks" ?

File: packages/operator/main.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+			if r.Method == http.MethodGet {
+				fileNames := getFilesInDirectory("./.platform/stacks", ".yaml")
+				var response []map[string]string
+	
+				for _, fileName := range fileNames {
+					id := strings.TrimSuffix(fileName, ".yaml")
+					filePath := filepath.Join("./.platform/stacks", fileName)
```
Comment (web7evgheni) [inline packages/operator/main.go:130]: @Timex1854 Please use environment variables for paths

File: packages/operator/main.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+					w.WriteHeader(http.StatusNotFound)
+					fmt.Fprintf(w, "Error reading file: %s", err)
+					return
+				}
+
+				if reqBody.Activate {
+					err = applyManifest(content, false)
+				} else {
+					err = applyManifest(content, true)
+				}
```
Comment (web7evgheni) [inline packages/operator/main.go:181]: @Timex1854 Replace with:
`err = applyManifest(content, reqBody.Activate)`

File: packages/operator/main.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+					fmt.Fprint(w, "Deactivated")
+				} else {
+					w.WriteHeader(http.StatusMethodNotAllowed)
+					fmt.Fprint(w, "Unsupported method")
+				}
+			})
+
+			corsWrappedHandler := corsHandler(http.DefaultServeMux)
+
+			if err := http.ListenAndServe(":8080", corsWrappedHandler); err != nil {
```
Comment (web7evgheni) [inline packages/operator/main.go:199]: @Timex1854 Please use environment variables for port

File: packages/operator/main.go [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+		fmt.Println("Operator is running...")
+
+		terminate := make(chan os.Signal, 1)
+		signal.Notify(terminate, syscall.SIGINT, syscall.SIGTERM)
+		<-terminate
+
+		fmt.Println("Received termination signal. Stopping the operator.")
+	}
+
+	func getFilesInDirectory(dirPath, suffix string) []string {
```
Comment (web7evgheni) [inline packages/operator/main.go:228]: @Timex1854 Main function always shall be last one.

File: packages/operator/main.go [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+	}
+
+	return "Not Activated"
+}
+
+
+
+
+
+
```
Comment (web7evgheni) [inline packages/operator/main.go:104]: @Timex1854 No Spaces

File: packages/operator/main.go [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+type Manifest struct {
+	Content []byte `json:"content"`
+}
+
+type ManifestMetadata struct {
+	Name string `yaml:"name"`
+}
+
+func applyManifest(content []byte, delete bool) error {
```
Comment (web7evgheni) [inline packages/operator/main.go:28]: @Timex1854 please add doc comments to all functions

**Review by web7evgheni — CHANGES_REQUESTED** [review]

File: packages/stack/stack-operator/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,10 @@
+FROM golang:1.20-alpine as build
+WORKDIR /app
+COPY . .
+RUN CGO_ENABLED=0 go build -o /operator
+
+FROM alpine/k8s:1.24.16
+WORKDIR /
+COPY --from=build /operator /
```
Comment (art10m) [inline packages/stack/stack-operator/Dockerfile:8]: @Timex1854 Use this approch for installing kubectl 
```suggestion
COPY --from=build /operator /
COPY --from=alpine/k8s:1.24.16 /usr/bin/kubectl /usr/local/bin/kubectl
```

File: packages/stack/stack-operator/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,10 @@
+FROM golang:1.20-alpine as build
+WORKDIR /app
```
Comment (art10m) [inline packages/stack/stack-operator/Dockerfile:2]:
```suggestion
WORKDIR /app
ENV STACKS_DIRECTORY=""
```

File: packages/stack/stack-operator/main.go [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+	}
+
+	return fileNames
+}
+// main is the entry point of the application.
+// It reads environment variables, starts the HTTP server, and watches the stacks directory for changes.
+func main() {
+	stacksDirectory := os.Getenv("STACKS_DIRECTORY")
+	port := os.Getenv("PORT")
+	if port == "" {
```
Comment (art10m) [inline packages/stack/stack-operator/main.go:128]: @Timex1854 Pls use ternary operator for cleaner code

Comment (Timex1854) [reply packages/stack/stack-operator/main.go]: ternary operator is absent in golang

File: packages/kndp/charts/kndp/Chart.yaml [hunk 11]
```diff
@@ -20,3 +20,9 @@ dependencies:
   - name: vault
     version: "~0.25.0"
     repository: https://helm.releases.hashicorp.com
+  - name: stack-operator
+    version: "~0.1.0"
+    repository: https://kndp.io
+  - name: cluster-operator
+    version: "~0.1.0"
+    repository: https://kndp.io
```
Comment (web7evgheni) [inline packages/kndp/charts/kndp/Chart.yaml:28]: @Timex1854 In this PR you working with stack operator, why you add cluster operator? Please remove it, it shall be added by other PR where you change cluster operator.

**Review by web7evgheni — CHANGES_REQUESTED** [review]

**Pull request merged**